code/_globalvars/regexes.dm

```diff
@@ -11,3 +11,7 @@ GLOBAL_DATUM_INIT(is_color_nocrunch, /regex, regex("^\[0-9a-fA-F]{6}$"))
 //All characters forbidden by filenames: ", \, \n, \t, /, ?, %, *, :, |, <, >, ..
 GLOBAL_DATUM_INIT(filename_forbidden_chars, /regex, regex(@{""|[\\\n\t/?%*:|<>]|\.\."}, "g"))
 GLOBAL_PROTECT(filename_forbidden_chars)
+
+//Banning panel inputs
+GLOBAL_DATUM_INIT(is_all_numbers, /regex, regex("^\[0-9]+$"))
```

Comment:
http://www.byond.com/docs/ref/#/proc/isnum

Reply:
I'll not have access to a Windows machine until friday :"

Reply:
Okay, yeah, my concerns were right, isnum checks if the variable type is a number, but this variable type is a string, so isnum will not work 
![image](https://github.com/BeeStation/BeeStation-Hornet/assets/45698448/80c48a16-4cef-4fc0-b619-d6c548ff7428)

Reply:
https://www.byond.com/docs/ref/#/proc/text2num

Reply:
If T is a text string for a number, return the number. Any non-numeric text following the initial portion will be ignored. If there is no initial numeric portion, the result is null. 

If I use that, I'll just get all the non-number characters gone...

Reply:
Which is exactly what you want. There shouldn't be any non-numeric values in the CID field. Ideally this would just be handled on the js side where the field itself only allows numeric entries.

Reply:
I actually did forgot about js number inputs (oops), ty Crossed!